browser/net/brave_reduce_language_network_delegate_helper.cc

```diff
@@ -63,11 +64,15 @@ std::string FarbleAcceptLanguageHeader(
   }
   // Add a fake q value after the language code.
   brave::FarblingPRNG prng;
-  if (g_brave_browser_process->brave_farbling_service()
-          ->MakePseudoRandomGeneratorForURL(
-              origin_url, profile && profile->IsOffTheRecord(), &prng)) {
+
+  auto* brave_farbling_service = BraveFarblingServiceFactory::GetForProfile(
+      profile->IsOffTheRecord() ? profile->GetOriginalProfile() : profile);
```

Comment:
you shouldn't pass `profile->GetOriginalProfile()` here. Just use `profile`, you need to configure service factory in a way that the service should understand that it's running under `OffTheRecordProfile`.

to do this pls use `ProfileKeyedServiceFactory` instead of `BrowserContextKeyedServiceFactory` and initialize the factory like so:
```
BraveFarblingServiceFactory::BraveFarblingServiceFactory()
    : ProfileKeyedServiceFactory(
          "BraveFarblingService",
          ProfileSelections::Builder()
              .WithRegular(ProfileSelection::kOwnInstance)
              .WithGuest(ProfileSelection::kOwnInstance)
              .WithSystem(ProfileSelection::kOwnInstance)
              .Build()) {}
```

this way the service should just have `session_token_`.

Reply:
ohh, I didn't know about this, Thanks!

Reply:
BTW, Do we really need to create the service for System profile?
@goodov

Reply:
For now, I'm not creating service for system profile.

Reply:
> For now, I'm not creating service for system profile.

as long as the `GetForProfile` result is checked before using, this is fine. I think the system profile is used on profile creation window and some other parts like this (when there's no profile yet).

Reply:
Yes, as per my knowledge it will used in profile picker as well.

@goodov Could you please help with CI for this PR?